src/features/surveys/components/SurveyEditor/blocks/OpenQuestionBlock.tsx

```diff
@@ -1,13 +1,222 @@
-import { FC } from 'react';
+import AbcIcon from '@mui/icons-material/Abc';
+import { FormattedMessage as Msg } from 'react-intl';
+import SortIcon from '@mui/icons-material/Sort';
+import theme from 'theme';
+import { useIntl } from 'react-intl';
 import { ZetkinTextQuestion } from 'utils/types/zetkin';
+import {
+  Box,
+  ClickAwayListener,
+  MenuItem,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { FC, useEffect, useRef, useState } from 'react';
 
 interface OpenQuestionBlockProps {
-  question: ZetkinTextQuestion;
+  element: ZetkinTextQuestion;
+  inEditMode: boolean;
+  onEditModeEnter: () => void;
+  onEditModeExit: (question: Omit<ZetkinTextQuestion, 'required'>) => void;
 }
 
-const OpenQuestionBlock: FC<OpenQuestionBlockProps> = ({ question }) => {
-  return <Box>{JSON.stringify(question)}</Box>;
+enum fieldType {
+  multiLine = 'multiLine',
+  singleLine = 'singleLine',
+}
+
+type elementWithoutRequired = Omit<ZetkinTextQuestion, 'required'>;
+
+const OpenQuestionBlock: FC<OpenQuestionBlockProps> = ({
+  element,
+  inEditMode,
+  onEditModeEnter,
+  onEditModeExit,
+}) => {
+  const intl = useIntl();
+
+  const [typeField, setTypeField] = useState(
+    element.response_config.multiline === true
+      ? fieldType.multiLine
+      : fieldType.singleLine
+  );
+
+  const [title, setTitle] = useState(element.question);
+  const [description, setDescription] = useState(element.description);
+
+  const [responseConfig, setResponseConfig] = useState(
+    element.response_config.multiline
+  );
+  const [focus, setFocus] = useState<
+    'title' | 'description' | 'responseConfig' | null
+  >(null);
+
+  const titleRef = useRef<HTMLInputElement>(null);
+  const descriptionRef = useRef<HTMLInputElement>(null);
+  const typeConfigRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (focus === 'title') {
+      const input = titleRef.current;
+      input?.focus();
+    } else if (focus === 'description') {
+      const input = descriptionRef.current;
+      input?.focus();
+    } else {
+      const input = typeConfigRef.current;
+      input?.focus();
+    }
+  }, [focus]);
+
+  const handleKeyDown = (evt: React.KeyboardEvent<HTMLDivElement>) => {
+    if (evt.key === 'Enter') {
+      const elementWithout: elementWithoutRequired = {
+        description: description,
+        question: title,
+        response_config: { multiline: responseConfig },
+        response_type: element.response_type,
+      };
+      onEditModeExit(elementWithout);
+      setFocus(null);
+    }
+  };
+
+  const handleSelect = (event: React.BaseSyntheticEvent) => {
+    setTypeField(event.target.value);
+  };
+
+  const createElementToUpdate = () => {
+    const elemenToUpdate: elementWithoutRequired = {
+      description: description,
+      question: title,
+      response_config: { multiline: responseConfig },
+      response_type: element.response_type,
+    };
+    return elemenToUpdate;
+  };
+
```

Comment:
I don't think I understand why this is a function that returns an object, instead of just creating the object?

Reply:
The element that returns is the one we need to send to the API to be saved, so it has a special format type. As I need to create this kind of element in several places, I´ve decided to have a "helper" function instead of have repeated code of the creation of the object, I guess is more a matter of style 😅

Reply:
As far as I can tell, the function is only used once, on line 101. Are you planning to, or did you previously use it in more locations, @rebecarubio?